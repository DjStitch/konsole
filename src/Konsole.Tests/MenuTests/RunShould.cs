using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using Konsole.Menus;
using NUnit.Framework;

namespace Konsole.Tests.MenuTests
{
    public class RunShould
    {
        [UseReporter(typeof(DiffReporter))]
        [Test]
        public void render_the_menu_with_default_item_selected_and_with_correct_colours()
        {
            var con = new MockConsole(15, 7);
            
            var m = new Menu(con, "MENU", ConsoleKey.Escape, 10,
                new MenuItem('a', "item 1", () => { }),
                new MenuItem('b', "item 2", () => { }),
                new MenuItem('c', "item 3", () => { })
                );
            m.Keyboard = new MockKeyboard(ConsoleKey.Escape);
            m.Run();

            var expected = new[]{
                "     MENU      ",
                "               ",
                "   a. item     ",
                "   b. item     ",
                "   c. item     ",
                "               ",
                "               ",
                };

            con.Buffer.ShouldBe(expected);

            expected = new[]{
                " wk wk yr yr yrMyrEyrNyrUyr yr yr yr wk wk wk",
                " wk wk wB wB wB wB wB wB wB wB wB wB wk wk wk",
                " wk wk wBawB.wB wBiBatBaeBamBa wB wB wk wk wk",
                " wk wk wBbwB.wB wBiwBtwBewBmwB wB wB wk wk wk",
                " wk wk wBcwB.wB wBiwBtwBewBmwB wB wB wk wk wk",
                " wk wk wB wB wB wB wB wB wB wB wB wB wk wk wk",
                " wk wk wk wk wk wk wk wk wk wk wk wk wk wk wk",
                };

            con.BufferWithColor.ShouldBe(expected);
        }


        [Test]
        public void render_the_menu_with_correct_theme_colours()
        {
            var con = new MockConsole(15, 7);

            var m = new Menu(con, "MENU", ConsoleKey.Escape, 10,
                new MenuItem('a', "item a", () => { }),
                new MenuItem('b', "item b", () => { }),
                new MenuItem('q', "item c", () => { })
                );
            m.Keyboard = new MockKeyboard(ConsoleKey.Escape);
            m.Run();

            var expected = new[]{
                " wk wk yr yr yrMyrEyrNyrUyr yr yr yr wk wk wk",
                " wk wk wB wB wB wB wB wB wB wB wB wB wk wk wk",
                " wk wk wBawB.wB wBiBatBaeBamBa wB wB wk wk wk",
                " wk wk wBbwB.wB wBiwBtwBewBmwB wB wB wk wk wk",
                " wk wk wBqwB.wB wBiwBtwBewBmwB wB wB wk wk wk",
                " wk wk wB wB wB wB wB wB wB wB wB wB wk wk wk",
                " wk wk wk wk wk wk wk wk wk wk wk wk wk wk wk",
                };

            con.BufferWithColor.ShouldBe(expected);
        }

    }
}
