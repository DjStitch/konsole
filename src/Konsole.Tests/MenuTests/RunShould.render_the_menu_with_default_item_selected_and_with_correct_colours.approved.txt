label[0] =  wk wk aB aB aB aB aB aB aB aB aB aB wk wk wk
label[1] =  wk wk aB aBMaBEaBNaBUaB aB aB aB aB wk wk wk
label[2] =  wk wk aB aB-aB-aB-aB-aB-aB-aB aB aB wk wk wk
label[3] =  wk wk aB aBiBatBaeBamBa Ba1Ba aB aB wk wk wk
label[4] =  wk wk aB aBiaBtaBeaBmaB aB2aB aB aB wk wk wk
label[5] =  wk wk aB aBiaBtaBeaBmaB aB3aB aB aB wk wk wk
label[6] =  wk wk aB aB aB aB aB aB aB aB aB aB wk wk wk
